benchmarks/definition/config.py

```diff
@@ -0,0 +1,45 @@
+"""Benchmark configurations."""
+
+from collections.abc import Callable
+from typing import TypeVar
+
+from attrs import define, field
+from attrs.validators import instance_of, is_callable
+from pandas import DataFrame
+
+from baybe.serialization.mixin import SerialMixin
+
+BenchmarkSettings = TypeVar("BenchmarkSettings")
+
+BenchmarkFunction = Callable[[BenchmarkSettings], DataFrame]
+
+
+@define(frozen=True)
+class BenchmarkFunctionDefinition:
+    """The core components including the benchmark function."""
+
+    callable: BenchmarkFunction = field(validator=is_callable())
+    """The callable that executes the benchmark code and returns the result."""
+
+    description: str = field(validator=instance_of(str))
+    """The description of the benchmark function."""
+
+    best_possible_result: float | None = field(default=None)
+    """The best possible result which can be achieved in the optimization process."""
+
+
+@define(frozen=True)
+class ConvergenceExperimentSettings(SerialMixin):
```

Comment:
I'm confused by this, this does not seem to be related to `BenchmarkSettings` in any way? I would have expected this to be a subclass of some general `BenchmarkSettings` class